tests/sentry/sentry_metrics/querying/data_v2/test_api.py

```diff
@@ -158,11 +160,19 @@ def test_query_with_empty_results(self) -> None:
             data = results["data"]
             assert len(data) == 1
             assert data[0][0]["by"] == {}
-            assert data[0][0]["series"] == [
-                None,
-                expected_identity_series,
-                expected_identity_series,
-            ]
+
+            matches_one = False
+            assert isinstance(expected_identity_series, list)  # mypy
+            for e in expected_identity_series:
+                if data[0][0]["series"] == [
+                    None,
+                    e,
+                    e,
+                ]:
+                    matches_one = True
+                    break
+            assert matches_one
```

Comment:
Why are we trying to match two values? Is it returned non-deterministically by snuba?

You could simplify this code in the following way:
```suggestion
            _, second, third = assert data[0][0]["series"]
            assert second in expected_identity_series
            assert third in expected_identity_series
```

I think this makes it more readable and reduces complexity in the tests. What do you think?

Reply:
> Why are we trying to match two values? Is it returned non-deterministically by snuba?

because the current value for `expected identity series` is
```
("sum", 0.0, 0.0),
("min", 0.0, 0.0)
```
but when the new query optimizer is running its
```
("sum", None, 0.0),
("min", None, 0.0)
```
so this change allows the test to pass before and after the new optimizer is deployed

> I think this makes it more readable and reduces complexity in the tests. What do you think?

I agree, this is much more readable, will make this change! Another option would be to xfail this test like we will do with the one below, and then remove xfail when the new query optimizer is deployed.

Reply:
Yeah, my idea was to put xfails and expect the new state. Having a test that can pass with both states might backfire in case we forget to refactor it.